pkg/profiler/profiler.go

```diff
@@ -748,11 +748,17 @@ func (p *CgroupProfiler) readKernelStack(kernelStackID int32, stack *stack) erro
 
 // readValue reads the value of the given key from the counts ebpf map.
 func (p *CgroupProfiler) readValue(keyBytes []byte) (uint64, error) {
-	valueBytes, err := p.bpfMaps.counts.GetValue(unsafe.Pointer(&keyBytes[0]))
-	if err != nil {
+	valueBytes := make([]byte, 8*runtime.NumCPU()) // u64 size (8 bytes) x NumCPU
```

Comment:
This looks fairly expensive to do for every single value, especially if we're now multiplying it by the number of CPUs. I'm happy to merge as is and base optimizations on profiling data though.